sympy/core/singleton.py

```diff
@@ -175,6 +175,9 @@ def __init__(cls, *args, **kwargs):
         cls.__new__ = lambda cls: obj
         cls.__getnewargs__ = lambda obj: ()
         cls.__getstate__ = lambda obj: None
+        def func(self):
+            return self.__class__
+        cls.func = property(func)
```

Comment:
Could this be `cls.func = property(lambda _: _.__class__)`?

Reply:
I think so. Do you think it would increase performance? Or is this considered to be better style? (possibly both)

Reply:
not sure about performance; just styl considering the surrounding code